cmds/webboot/webboot.go

```diff
@@ -129,22 +129,24 @@ func main() {
 	}
 
 	cachedDir := *dir
-
 	if cachedDir == "" {
 		mp, err := getCachedDirectory()
-		if err != nil {
-			log.Fatalf("Fail to find the USB stick: %+v", err)
+		if err == nil {
+			cachedDir = filepath.Join(mp.Path, "Image")
 		}
-		cachedDir = filepath.Join(mp.Path, "Image")
 	}
 	entries := []menu.Entry{
```

Comment:
What if you created entries before checking cachedDir, and then if cachedDir exists, append the option to entries - this way you don't have to check it twice

Reply:
the problem is, the cached dir not only could be the cache dir in the USB, but also could be a local dir given by args, so if I insert entry here, I should also make another if conditional sentance that is when dir args is not empty. It is not so simple too